### PR TITLE
Fixes ads service will crash if a user previously had ads history and triggers OnUnIdle before Brave ads are initialized

### PR DESCRIPTION
**Pull request by tmancey**
Fixes https://github.com/brave/brave-browser/issues/6880

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan:

See https://github.com/brave/brave-browser/issues/6880 for steps

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

**Comment by tmancey** [comment]
Windows failed CI so restarting Windows build

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc [hunk 1]
```diff
@@ -231,7 +231,7 @@ bool AdsImpl::IsInitialized() {
 void AdsImpl::Shutdown(
     ShutdownCallback callback) {
   if (!is_initialized_) {
-    BLOG(WARNING) << "Failed to shutdown ads as not initialized";
+    BLOG(WARNING) << "Shutdown failed as not initialized";
```
Comment (masparrow) [inline vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc]: Better as (INFO)? It's fair to say you can't shut something down if it's not initialised, so really this is just diagnostic at this point to say something called it prematurely?

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc [hunk 2]
```diff
@@ -342,6 +344,10 @@ void AdsImpl::OnUnIdle() {
 
   BLOG(INFO) << "Browser state changed to unidle";
 
+  if (!IsInitialized()) {
+    return;
```
Comment (masparrow) [inline vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc]: BLOG here perhaps? Or superfluous?

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc (hunk 1 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc]: Our logs are quite busy as is, so wanted to reduce noise. When diagnosing issues we use a higher verbosity.

File: vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc (hunk 2 above)
Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc]: Good point, adding now

Comment (tmancey) [reply vendor/bat-native-ads/src/bat/ads/internal/ads_impl.cc]: Fixed

**Pull request merged**